### PR TITLE
Fixing problem rendering Lato font

### LINKED ISSUE #3461: Lato font rendering is not working properly

### PR DESCRIPTION
**Pull request by xavijam**
We have a really big problem rendering Lato font with our font generated files. So @matallo has suggested to use [webfontloader](https://github.com/typekit/webfontloader) for loading them from Google, and if they aren't loaded, load the local ones (perfect for on-premise case). I feel it is a good option, what do you think?

@CartoDB/frontend 

Fixes #3461.

### CODE REVIEW

**Comment by xavijam** [comment]
I haven't done anything yet, this it for discussion.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3409/))

**Comment by xavijam** [comment]
@matallo approach done, @viddo could you check it?

File: lib/assets/javascripts/cartodb/fonts/lato_loader.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var customInstall = cdb.config.get('cartodb_com_hosted');
+var latoCSSUrl = cdb.config.get('assets_url') + '/stylesheets/fonts.css';
+var opts = {};
+
+if (customInstall) {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/fonts/lato_loader.js]: custom install, onpremise,...

File: lib/assets/javascripts/cartodb/fonts/lato_loader.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  opts = {
+    custom: {
+      families: ['Lato'],
+      urls: [ latoCSSUrl ]
+    }
+  }
+} else {
+  opts = {
+    fontinactive: function () {
+      WebFont.load({
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb/fonts/lato_loader.js]: If font loading fails, let's load local fonts.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3412/))

**Comment by viddo** [comment]
Solution itself looks legit, but why is it that the font loading fails when loaded "locally" (CDN?) in the first place?

**Comment by xavijam** [comment]
I don't get you @viddo sorry. fontinactive is an event triggered when the fonts haven't been loaded, then we load the local ones. They only are loaded from the beginning if the editor is a "custom_install" type.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3419/))

**Comment by alonsogarciapablo** [comment]
I'm personally missing some context here too. As @viddo said, solution looks good, but I'm not sure what the real problem is.

**Comment by matallo** [comment]
I find this awesome, good job @xavijam

**Comment by xavijam** [comment]
@alonsogarciapablo We had problems rendering Lato font using it locally, before being loaded, weird characters appeared in the texts with that font applied (unfortunately i don't have an screenshot right now). After checking what we do in CartoDB website, I feel this way is way better, and without forgetting open source and offline users.

**Comment by Cartofante** [comment]
Frontend tests were OK :+1: ([details](http://clinker.cartodb.net/jenkins/job/CartoDB-PR-testing/3425/))

**Comment by alonsogarciapablo** [comment]
@xavijam I guess you're referring to this:

image

Thanks for the context and the quick fix!

**Comment by xavijam** [comment]
Yep, that thing :(


On 4 May 2015 at 15:26:18, Pablo Alonso (notifications@github.com) wrote:

@xavijam I guess you're referring to this:



Thanks for the context and the quick fix!

**Pull request merged**